### DISCUSSION TITLE
test_fund_me.py failed on mainnet-fork-dev and run well on ganache-local

### DISCUSSION THREAD

**Discussion by frank0532** [smartcontractkit/full-blockchain-solidity-course-py#1677] [Q&A]
Code is as below:
![image](https://user-images.githubusercontent.com/47261346/187053930-96d1f444-b41d-4940-aeb0-3984a202ace5.png)

Get Error:
Brownie v1.19.1 - Python development framework for Ethereum

========================================================================= test session starts ==========================================================================
platform win32 -- Python 3.9.6, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: D:\syn\web3\brownie_fund_me
plugins: eth-brownie-1.19.1, hypothesis-6.27.3, forked-1.4.0, xdist-1.34.0, web3-5.30.0
collected 2 items
Attached to local RPC client listening at '127.0.0.1:8545'...

tests\test_fund_me.py Fs                                                                                                                                          [100%]

=============================================================================== FAILURES =============================================================================== 
______________________________________________________________________ test_can_fund_and_withdraw ______________________________________________________________________

    def test_can_fund_and_withdraw():
        account = get_account()
        fund_me = deploy_fund_me()
>       entrance_fee = fund_me.getEntranceFee() + 100

tests\test_fund_me.py:11:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _  
C:\Users\MiC\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\multicall.py:115: in _proxy_call
    result = ContractCall.__call__(*args, **kwargs)  # type: ignore
C:\Users\MiC\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\contract.py:1902: in __call__
    return self.call(*args, block_identifier=block_identifier, override=override)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _  

self = <ContractCall 'getEntranceFee()'>, block_identifier = None, override = None, args = ()
tx = {'allow_revert': None, 'data': '0x09bc33a7', 'from': '0x58a78Af7b613BD286328759A2d189e8884e4fA1C', 'gas': None, ...}

    def call(
        self, *args: Tuple, block_identifier: Union[int, str, bytes] = None, override: Dict = None
    ) -> Any:
        """
        Call the contract method without broadcasting a transaction.

        Arguments
        ---------
        *args
            Contract method inputs. You can optionally provide a
            dictionary of transaction properties as the last arg.
        block_identifier : int | str | bytes, optional
            A block number or hash that the call is executed at. If not given, the
            latest block used. Raises `ValueError` if this value is too far in the
            past and you are not using an archival node.
        override : dict, optional
            A mapping from addresses to balance, nonce, code, state, stateDiff
            overrides for the context of the call.

        Returns
        -------
            Contract method return value(s).
        """

        args, tx = _get_tx(self._owner, args)
        if tx["from"]:
            tx["from"] = str(tx["from"])
        del tx["required_confs"]
        tx.update({"to": self._address, "data": self.encode_input(*args)})

        try:
            data = web3.eth.call({k: v for k, v in tx.items() if v}, block_identifier, override)
        except ValueError as e:
>           raise VirtualMachineError(e) from None
E           brownie.exceptions.VirtualMachineError: revert

C:\Users\MiC\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\contract.py:1693: VirtualMachineError
------------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------- 
mainnet-fork-dev
Contract deployed to 0x918Cf92680C6995aAfEc4e3E0AA6036b1540445D
=========================================================================== warnings summary =========================================================================== 
C:\Users\MiC\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\main.py:44
  C:\Users\MiC\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\main.py:44: BrownieEnvironmentWarning: Development network has a block height of 70       
    warnings.warn(

tests/test_fund_me.py::test_can_fund_and_withdraw
tests/test_fund_me.py::test_can_fund_and_withdraw
  C:\Users\MiC\.local\pipx\venvs\eth-brownie\lib\site-packages\eth_abi\codec.py:87: DeprecationWarning: abi.encode_abi() and abi.encode_abi_packed() are deprecated and will be removed in version 4.0.0 in favor of abi.encode() and abi.encode_packed(), respectively
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================================================================= short test summary info ======================================================================== 
FAILED tests/test_fund_me.py::test_can_fund_and_withdraw - brownie.exceptions.VirtualMachineError: revert
=============================================================== 1 failed, 1 skipped, 3 warnings in 2.20s ===============================================================

**Comment by cromewar** [comment]
Hello @frank0532 
this might be to several reasons so please check the following:
1. you have added correctly mainnet-fork-dev to your brownie config, without any typo or error.
2. You have add correctly the network to brownie at `brownie networks add` with correct rpcserver.
3. A extra recommendation, do not use pipx for brownie, is better to use regular pip to install it.
